src/ui/SmartSnippet/SmartSnippet.ts

```diff
@@ -8,8 +9,36 @@ import 'styling/_SmartSnippet';
 import { find } from 'underscore';
 import { IQueryResult } from '../../rest/QueryResult';
 import { UserFeedbackBanner } from './UserFeedbackBanner';
-import { analyticsActionCauseList, IAnalyticsNoMeta, IAnalyticsSmartSnippetContentLink } from '../Analytics/AnalyticsActionListMeta';
+import {
+  analyticsActionCauseList,
+  IAnalyticsNoMeta,
+  IAnalyticsSmartSnippetContentLinkMeta,
+  IAnalyticsSmartSnippetExplainWhyMeta,
+  IAnalyticsSmartSnippetExplainWhyDetailedMeta
+} from '../Analytics/AnalyticsActionListMeta';
 import { HeightLimiter } from './HeightLimiter';
+import { ExplanationModal, IExplanation } from './ExplanationModal';
+import { l } from '../../strings/Strings';
+
+interface ISmartSnippetExplaination {
+  analyticsId: string;
```

Comment:
Similar to an earlier comment, we could type `analyticsId` more strictly here.

Reply:
I'm also going to rename it `analytics`